src/lib.rs

```diff
@@ -33,6 +33,7 @@
 //! # Ok(())
 //! # }
 //! ```
+#![forbid(unsafe_code)]
```

Comment:
When dealing with threads, lifetimes, especially when trying to be greedy with bounds like this very PR of mine does, knowing that no `unsafe_code` is involved is quite reassuring 😇